Index: ChangeLog
===================================================================
--- ChangeLog	(Revision 5374)
+++ ChangeLog	(Arbeitskopie)
@@ -1,3 +1,7 @@
+2009-04-29  Oliver Hader  <oliver@typo3.org>
+
+	* Added feature #10869: Integrate hook to pre-process the frontend rendering
+
 2009-04-28  Stanislas Rolland  <typo3@sjbr.ca>
 
 	* Fixed bug #11003: htmlArea RTE: empty mutuallyExclusiveClasses array may cause js error
Index: typo3/sysext/cms/tslib/index_ts.php
===================================================================
--- typo3/sysext/cms/tslib/index_ts.php	(Revision 5374)
+++ typo3/sysext/cms/tslib/index_ts.php	(Arbeitskopie)
@@ -133,6 +133,17 @@
 }
 
 
+// Hook to preprocess the current request:
+if (is_array($TYPO3_CONF_VARS['SC_OPTIONS']['tslib/index_ts.php']['preprocessRequest'])) {
+	foreach ($TYPO3_CONF_VARS['SC_OPTIONS']['tslib/index_ts.php']['preprocessRequest'] as $hookFunction) {
+		$hookParameters = array();
+		t3lib_div::callUserFunction($hookFunction, $hookParameters, $hookParameters);
+	}
+	unset($hookFunction);
+	unset($hookParameters);
+}
+
+
 // *********************
 // Look for extension ID which will launch alternative output engine
 // *********************
